server/server_test.go

```diff
@@ -838,6 +838,144 @@ func (cli *testServerClient) runTestLoadData(c *C, server *Server) {
 
 		dbt.mustExec("drop table if exists pn")
 	})
+
+	err = fp.Close()
+	c.Assert(err, IsNil)
+	err = os.Remove(path)
+	c.Assert(err, IsNil)
+
+	fp, err = os.Create(path)
+	c.Assert(err, IsNil)
+	c.Assert(fp, NotNil)
+
+	// Test Column List Specification
+	_, err = fp.WriteString(
+		`1,2` + "\n" +
+			`3,4` + "\n")
+	c.Assert(err, IsNil)
+
+	cli.runTestsOnNewDB(c, func(config *mysql.Config) {
+		config.AllowAllFiles = true
+		config.Params = map[string]string{"sql_mode": "''"}
+	}, "LoadData", func(dbt *DBTest) {
+		dbt.mustExec("drop table if exists pn")
+		dbt.mustExec("create table pn (c1 int, c2 int)")
+		dbt.mustExec("set @@tidb_dml_batch_size = 1")
+		_, err1 := dbt.db.Exec(`load data local infile '/tmp/load_data_test.csv' into table pn FIELDS TERMINATED BY ',' (c1, c2)`)
+		dbt.Assert(err1, IsNil)
+		var (
+			a int
+			b int
+		)
+		rows := dbt.mustQuery("select * from pn")
+		dbt.Check(rows.Next(), IsTrue, Commentf("unexpected data"))
+		err = rows.Scan(&a, &b)
+		dbt.Check(err, IsNil)
+		dbt.Check(a, Equals, 1)
+		dbt.Check(b, Equals, 2)
+		dbt.Check(rows.Next(), IsTrue, Commentf("unexpected data"))
+		err = rows.Scan(&a, &b)
+		dbt.Check(err, IsNil)
+		dbt.Check(a, Equals, 3)
+		dbt.Check(b, Equals, 4)
+		dbt.Check(rows.Next(), IsFalse, Commentf("unexpected data"))
+
+		dbt.mustExec("drop table if exists pn")
+	})
+
+	err = fp.Close()
+	c.Assert(err, IsNil)
+	err = os.Remove(path)
+	c.Assert(err, IsNil)
+
+	fp, err = os.Create(path)
+	c.Assert(err, IsNil)
+	c.Assert(fp, NotNil)
+
+	// Test Column List Specification
+	_, err = fp.WriteString(
+		`1,2,3` + "\n" +
+			`4,5,6` + "\n")
+	c.Assert(err, IsNil)
+
+	cli.runTestsOnNewDB(c, func(config *mysql.Config) {
+		config.AllowAllFiles = true
+		config.Params = map[string]string{"sql_mode": "''"}
+	}, "LoadData", func(dbt *DBTest) {
+		dbt.mustExec("drop table if exists pn")
+		dbt.mustExec("create table pn (c1 int, c2 int, c3 int)")
+		dbt.mustExec("set @@tidb_dml_batch_size = 1")
+		_, err1 := dbt.db.Exec(`load data local infile '/tmp/load_data_test.csv' into table pn FIELDS TERMINATED BY ',' (c1, @dummy)`)
```

Comment:
@gentcys hi, please help fix CI, it seems this test paniced

```
[2020/07/28 12:05:45.934 +08:00] [ERROR] [conn.go:1109] ["process routine panicked"] [conn=10] [r={}] [stack="github.com/pingcap/tidb/server.processStream.func1\n\t/home/robi/Code/go/src/github.com/pingcap/tidb/server/conn.go:1111\nruntime.gopanic\n\t/home/robi/Runtime/go/src/runtime/panic.go:679\nruntime.goPanicIndex\n\t/home/robi/Runtime/go/src/runtime/panic.go:75\ngithub.com/pingcap/tidb/executor.(*InsertValues).getRow\n\t/home/robi/Code/go/src/github.com/pingcap/tidb/executor/insert_common.go:477\ngithub.com/pingcap/tidb/executor.(*LoadDataInfo).colsToRow\n\t/home/robi/Code/go/src/github.com/pingcap/tidb/executor/load_data.go:596\ngithub.com/pingcap/tidb/executor.(*LoadDataInfo).InsertData\n\t/home/robi/Code/go/src/github.com/pingcap/tidb/executor/load_data.go:516\ngithub.com/pingcap/tidb/server.insertDataWithCommit\n\t/home/robi/Code/go/src/github.com/pingcap/tidb/server/conn.go:1083\ngithub.com/pingcap/tidb/server.processStream\n\t/home/robi/Code/go/src/github.com/pingcap/tidb/server/conn.go:1144"]
```

you also can reproduce it in local machine by running those commands in shell:

1. `make failpoint-enable`
2. `go test ./server --check.f=TestLoadData`